AK/String.h

```diff
@@ -57,6 +57,9 @@ namespace AK {
 
 class String {
 public:
+    using iterator = char*;
+    using const_iterator = const char*;
```

Comment:
Please call these `Iterator` and `ConstIterator` like all other types in the system :)